### PR TITLE
fix(agents): sidebar scrollbar, pane-picker overlap, Global Assistant in drive spawn

### PR DESCRIPTION
**Pull request by 2witstudios**
## Summary
- Hide the Agents sidebar's session-list scrollbar entirely (new `.scrollbar-none` utility) instead of the Radix `ScrollArea`'s always-visible overlay bar, which collided with row "+" buttons.
- Fix `PanePicker`'s split-pane overlap: its sub-list wrappers used `min-h-0` on non-scrolling flex boxes, letting a short pane crush them while their `shrink-0` buttons refused to shrink — pushing "Global Assistant" past its box and stealing clicks from "Agents" below. Switched to `shrink-0`.
- Offer Global Assistant as the first pick in a drive's "+" spawn palette, landing the new session under that drive (loosened the session-spawn validation gate, which previously only allowed a driveless assistant spawn even though drive-scoped assistant conversations were already supported everywhere else — via the pane picker and the access/conversation layers).
- Default naming: "Assistant" → "Global Assistant" for the auto-named session label, the naming-step placeholder, and the expanded-conversation fallback.
- **(Review follow-up)** Thread the session's own `driveId` into the Global Assistant's chat request context instead of deriving it from the pathname. Turned out `agents` isn't a route `buildContextRef`/`tab-title.ts` recognizes as drive-scoped at all, so a Global Assistant conversation inside an agent session never got its drive's prompt/page-tree scope from any entry point (pre-existing, not introduced by this PR, but the new spawn-palette path made it reachable in one more way). See the review thread for detail.
- Added a `CHANGELOG.md` entry (manual, matching this repo's convention — the `changelog:generate` script is known-stale).

## Test plan
- [x] `bun run typecheck` (apps/web) — clean
- [x] `route.test.ts` — 34/34 passing, including new tests for the drive-scoped assistant spawn and the narrowed validation gate
- [x] `AgentsSidebar.test.tsx` — 48/48 passing, including a new test for picking Global Assistant from a drive's "+"
- [x] `PanePicker.test.tsx`, `AgentPanes.test.tsx`, `SessionPanes.test.tsx` — 65/65 passing
- [x] `useAssistantSessionChat.test.ts` (new) — covers the driveId-override and the driveless fallback
- [x] `bun run test:unit` (full monorepo lib + web suite) — 15502/15503 passing; the 1 failure (`grouping.test.ts` midnight-boundary case) is a pre-existing, unrelated TZ-environment flake (fails under local CDT, passes under UTC), not touched by this PR
- [x] `bun run test:security` — 44/50 suites passing; the 6 failures match this repo's known pre-existing baseline (deleted Login/Signup/Mobile-Login routes, excluded Session Service/Device Auth/Permissions suites) — confirmed identical failure set, nothing new
- [x] `bun run lint` (repo-wide build+lint) — clean
- [x] Verified no merge conflicts / branch fully up to date with `master`
- [ ] Manual: confirm no scrollbar renders on `/dashboard/agents`'s left sidebar, at rest or on hover
- [ ] Manual: split a pane and confirm "Global Assistant" is clickable and doesn't collide with "Agents"
- [ ] Manual: click "+" on a drive, pick Global Assistant, confirm the new session lands under that drive
- [ ] Manual: from the global `/dashboard/agents` view, spawn Global Assistant for a specific drive and confirm the assistant has that drive's context (page tree / system prompt)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01MMC7fGVwuwyD7NChEaQobP

## Summary by CodeRabbit

- **New Features**
  - Start Global Assistant sessions scoped to a specific drive.
  - Global Assistant now uses the associated drive for conversation context.
  - Global sessions are named “Global Assistant” by default.

- **Bug Fixes**
  - Improved pane picker spacing so options remain accessible in smaller panes.
  - Hidden scrollbars in the Agents session list while preserving scrolling.
  - Fixed drive context when viewing a session from a different drive.

### CODE REVIEW

File: apps/web/src/components/layout/left-sidebar/AgentsSidebar.tsx [hunk 1]
```diff
@@ -829,6 +828,14 @@ function SpawnSessionPalette({
           <CommandInput placeholder="Search agents…" autoFocus />
           <CommandList>
             <CommandGroup>
+              <CommandItem
+                value="assistant-Global Assistant"
+                disabled={spawning}
+                onSelect={() => onPickTarget({ kind: 'assistant', agentPageId: null, label: 'Global Assistant' })}
```
[the remark this answers is not in the document]
Comment (2witstudios) [reply apps/web/src/components/layout/left-sidebar/AgentsSidebar.tsx:834]: Confirmed and fixed in 92b52a814. The gap was actually bigger than described: `agents` isn't in `tab-title.ts`'s `DRIVE_SPECIAL_ROUTES` at all, so `buildContextRef` has never resolved a drive for *any* Global Assistant conversation inside an agent session (drive-scoped or global) reached from the Agents console — this predates this PR and also affects the pre-existing split-pane-picker path, not just the new spawn-palette entry point.

Fix: threaded the session's own `driveId` (already available on `AgentPanes`) through `PaneChat` -> `AssistantSessionChat` -> `useAssistantSessionChat`, and use it directly (`{ routeType: drive, driveId }`) instead of parsing the pathname whenever the session has a drive. A driveless global-assistant session keeps the pathname-derived context (unaffected — there's no session driveId to prefer there). Added `useAssistantSessionChat.test.ts` covering both the override and the fallback. Left `useAgentSessionChat` untouched: an agent conversation's drive is already anchored server-side via its `agentPageId` regardless of `contextRef`, so the same pathname imprecision there only affects incidental "this page" references, not core drive scoping — a smaller, separate, pre-existing concern I left out of scope for this PR.

Leaving this thread open for verification rather than resolving it myself.
